src/librustc/middle/ty/mod.rs

```diff
@@ -1196,11 +1197,18 @@ impl<'a, 'tcx> ParameterEnvironment<'a, 'tcx> {
                             MethodTraitItem(ref method_ty) => {
                                 let method_generics = &method_ty.generics;
                                 let method_bounds = &method_ty.predicates;
+                                let extent = if id != body_id {
```

Comment:
I suspect `body_id` is not used elsewhere... probably cleaner to write `body.is_none()` or `match body` here.